portfolio/src/main/webapp/script.js

```diff
@@ -36,8 +36,22 @@ function myFunction() {
   document.getElementById("myBar").style.width = scrolled + "%";
 }
 
-async function getMessage() {
-    const response = await fetch('/data');
-    const message = await response.text();
-    document.getElementById('message-container').innerHTML = message;
+function getMessage() {
+    fetch('/data').then(response => response.json()).then((someMessages) => {
+
+        const commentsListElement = document.getElementById('message-container');
+        commentsListElement.innerHTML = '';
+        commentsListElement.appendChild(
```

Comment:
If you are going to use the DOM manipulation routines (good!) then maybe don't assign a string to `innerHTML` as that forces a trip through the parser: you can loop over the children of `commentsListElement` and call `remove` on them.

Reply:
Honestly, I saw it being implemented that way on the tutorial and thought of it as effective, since I saw it as a good way of starting the html element first as empty and then add the hard-coded messages to it.